### ISSUE TITLE
config-write-py includes temporary settings

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#5907]
After a `tsh` with a clear basedir and `:config-write-py`, I see:

```python
config.set('content.javascript.enabled', True, '*://start.duckduckgo.com/*')
```

in the generated `config.py`.

**Comment by t-ens** [comment]
I think the problem here and also in issue #5890 with config-write-py including internal settings is that in the `config_write_py` function in `configcommands.py`, line 466 is `for values in self._config` and self._config contains all settings including temporary and internal settings.

As far as I can tell the only difference between setting a variable with the temporary argument and without it is that without the temporary argument the setting gets added to YAML as well as the main config so am I correct in thinking that line 466 should be `for values in self._config._yaml` instead?

If I make this change temporary settings are no longer written and the internal settings are not either however I did notice that if I start qutebrowser with an already existing config.py, and then issue `config-write-py --force` none of the settings from the original config.py will be written to the new config.py. Is this what should happen or would  I also need to change the code which reads config.py to add all of these settings to YAML as well?

**Comment by The-Compiler** [comment]
I don't think the solution for #5890 and this needs to be the same necessarily. We already have the `hide_userconfig` flag for internal settings so they aren't displayed in e.g. `:config-diff` (5985b474c828b46d5297644b748f434d91211690) - a possible solution for #5890 would be to ignore values with that flag set.

However, you make a good point about the possibility of `:config-write-py` taking the changed settings from the `YamlConfig` object instead. Like you say, that'd be a (rather subtle) behavior change if a `config.py` was read previously.

I don't see a strong reason against doing things this way, though I wonder what others think. I'm actually not sure what the usecases are for `:config-write-py` without `--defaults`, I suppose most people use it *with* `--defaults` to get a `config.py` as a starting point without any previous configuration.

**Comment by t-ens** [comment]
I don't understand the code well enough to tell whether it would be better to use the YamlConfig object or instead do something like introducing a `temporary` flag similar to `hide_userconfig` and ignore any setting with this flag set but I can give implementing whichever you think is better a go.

I don't personally do it this way either, but the thing I was worried about is that if someone has a customized `config.py`, then maybe changes some setting while using qutebrowser and uses `:config-write-py --force` intending to update their `config.py` they may be surprised that their whole customized `config.py` is now gone except for the new setting that was just set in the browser. Maybe no one actually tries to use it this way and it is not an issue though.

**Comment by appleseedexm** [comment]
@t-ens I am actually using it this way but I'm fairly new to qute and not really sure if I'm misunderstanding something. 
My workflow now is to basically `:config-write-py --force` the config all the time when I change a setting directly from qute, which is my usual usecase, but sometimes I edit the `config.py` directly, and when I do that I usually source and then write again to make it conform to the way qute writes the config. 

The problem I started facing is with my second configuration (for work), that sources my main config. I've yet to find a solution to stop qute from removing my `config.source(path/to/main/conf.py)` whenever I want to `:config-write-py --force` the config in my work setup (e.g. when I want to change work specific settings directly from qute).

**Comment by The-Compiler** [comment]
@appleseedexm Maybe write to a different file instead (`:config-write-py` takes a filename)? But in any case, this is completely unrelated to this issue.

**Comment by appleseedexm** [comment]
Yes, sorry, found this by accident and wanted to give context that some people _are_ in fact using it this way, my comment was not meant as an issue.
